spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/CoverController.java

```diff
@@ -43,51 +45,16 @@
 import org.springframework.web.multipart.MultipartFile;
 import org.springframework.web.servlet.ModelAndView;
 
+@Slf4j
 @Controller
 @RequestMapping(value = "/admin/covers")
 public class CoverController {
 
-    private static final Logger log = LoggerFactory.getLogger(CoverController.class);
-
-    @Autowired
-    private CoverRepository coverRepository;
-
-    @Autowired
-    private ServiceApiRepository serviceRepository;
-
-    @Autowired
-    private CoverImageReadService coverImageReadService;
-
     @Autowired
-    AdminServicesController adminServicesController;
-
-    @Value("${ygg.admin.servicecovers.path}")
-    private String serviceCoversPath;
-
-    @Value("${ygg.admin.servicecovers.url}")
-    private String serviceCoversUrl;
+    private CoverService coverService;
 
     @RequestMapping(value = "/{id}/upload")
-    public ModelAndView fileUpload(Authentication auth, @PathVariable("id") Long serviceId, @RequestParam("file") MultipartFile file) throws IOException, ParseException, java.text.ParseException {
-
-        ServiceApi service = serviceRepository.findById(serviceId).orElse(null);
-
-        String coverFilename = UUID.randomUUID().toString();
-
-        ServiceCover cover = new ServiceCover();
-        cover.setService(service);
-        cover.setOriginalFileName(StringUtils.cleanPath(file.getOriginalFilename()));
-        cover.setFileName(coverFilename);
-        cover.setFileType(file.getContentType());
-        cover.setSize(file.getSize());
-        cover.setLocalPath(serviceCoversPath + File.separator + coverFilename);
-        cover.setUrl(serviceCoversUrl + coverFilename);
-
-        coverImageReadService.readExcelInputStream(new ByteArrayInputStream(file.getBytes()),
-                new File(cover.getLocalPath()));
-
-        coverRepository.save(cover);
-
-        return adminServicesController.serviceListPage();
+    public ModelAndView fileUpload(Authentication auth, @PathVariable("id") Long serviceId, @RequestParam("file") MultipartFile file) throws IOException, java.text.ParseException {
+        return coverService.fileUpload(auth, serviceId, file);
```

Comment:
```suggestion
        return coverService.uploadFile(auth, serviceId, file);
```

Reply:
I find the names "CoverService/Controller" rather unexpressive. Please find a better name.